src/marshmallow/context.py

```diff
@@ -0,0 +1,25 @@
+"""Objects related to serializtion/deserialization context"""
+
+import contextlib
+import contextvars
+
+from marshmallow.utils import no_default
+
+
+class Context(contextlib.AbstractContextManager):
```

Comment:
thought: should we also expose API for setting the context? 

```python
from marshmallow import Context

@app.before_request
def bind_marshmallow_context():
    Context.set({"user_id": current_user.id})
```

Reply:
Hmmm. I think this is meant to be used as a context manager, so that the context is reset on exit.

We'd need to reset in `@app.after_request` or something. And manage the request crash case. Things context managers do.

I'm not sure how I'd deal with this use case, though.